skyvern/forge/sdk/workflow/context_manager.py

```diff
@@ -195,30 +201,92 @@ async def register_secret_workflow_parameter_value(
 
         LOG.info(f"Fetching credential parameter value for credential: {credential_id}")
 
-        db_credential = await app.DATABASE.get_credential(credential_id, organization_id=organization.organization_id)
-        if db_credential is None:
-            raise CredentialParameterNotFoundError(credential_id)
+        # Handle 1Password references directly (op://vault_id/item_id/field)
+        if credential_id.startswith("op://"):
+            try:
+                # Use the 1Password SDK to resolve the reference directly
+                secret_value = await resolve_secret(credential_id)
+                random_secret_id = self.generate_random_secret_id()
+                self.secrets[random_secret_id] = secret_value
+                self.values[parameter.key] = random_secret_id
+                self.parameters[parameter.key] = parameter
+                return
+            except Exception as e:
+                LOG.error(f"Failed to resolve 1Password reference: {credential_id}. Error: {e}")
+                raise e
+
+        # Handle 1Password credentials in vault_id:item_id format
+        if ":" in credential_id and not credential_id.startswith("op://"):
+            try:
+                LOG.info(f"Processing 1Password credential in vault_id:item_id format: {credential_id}")
+                # Use the 1Password SDK to resolve the reference
+                import json
+
+                secret_value_json = await resolve_secret(credential_id)
+
+                # Validate the JSON response
+                if not secret_value_json:
+                    LOG.error(f"Empty response from 1Password for credential: {credential_id}")
+                    raise ValueError(f"Empty response from 1Password for credential: {credential_id}")
+
+                try:
+                    secret_values = json.loads(secret_value_json)
+                except json.JSONDecodeError as json_err:
+                    LOG.error(f"Invalid JSON response from 1Password: {secret_value_json[:100]}... Error: {json_err}")
+                    raise ValueError(f"Invalid JSON response from 1Password: {json_err}")
+
+                if not secret_values:
+                    LOG.warning(f"No values found in 1Password item: {credential_id}")
+                    # Still continue with empty values
 
-        bitwarden_credential = await BitwardenService.get_credential_item(db_credential.item_id)
+                self.parameters[parameter.key] = parameter
+                self.values[parameter.key] = {}
 
-        credential_item = bitwarden_credential.credential
+                # Process each field in the 1Password item
+                for key, value in secret_values.items():
```

Comment:
if the "key" is for the otp secret, let's treat this differently. see how bitwarden does for the TOTP thing